server/settings.go

```diff
@@ -2,24 +2,44 @@ package main
 
 import (
 	"github.com/mattermost/mattermost-server/v6/model"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
 
 	"github.com/mattermost/mattermost-plugin-jira/server/utils/types"
 )
 
 const (
 	settingOn  = "on"
 	settingOff = "off"
+
+	errStoreNewSettings = "Could not store new settings. Please contact your system administrator. Error: %v"
+	errConnectToJira    = "Your account is not connected to Jira. Please type `/jira connect`. %v"
+
+	assigneeRole = "assignee"
+	mentionRole  = "mention"
+	reporterRole = "reporter"
+	watchingRole = "watching"
 )
 
+func (connection *Connection) updateRolesForDMNotification(role string, hasNotification bool) bool {
+	if role != assigneeRole && role != mentionRole && role != reporterRole && role != watchingRole {
+		return false
+	}
+	if connection.Settings.RolesForDMNotification == nil {
+		connection.Settings.RolesForDMNotification = make(map[string]bool)
+	}
+	connection.Settings.RolesForDMNotification[role] = hasNotification
+	return true
+}
 func (p *Plugin) settingsNotifications(header *model.CommandArgs, instanceID, mattermostUserID types.ID, connection *Connection, args []string) *model.CommandResponse {
-	const helpText = "`/jira settings notifications [value]`\n* Invalid value. Accepted values are: `on` or `off`."
+	const helpText = "`/jira settings notifications [assignee|mention|reporter|watching] [value]`\n* Invalid value. Accepted values are: `on` or `off`."
 
-	if len(args) != 2 {
+	if len(args) != 3 {
 		return p.responsef(header, helpText)
 	}
 
 	var value bool
-	switch args[1] {
+	switch args[2] {
```

Comment:
These magic numbers would be much better in a constant, so they're named in a descriptive way.

https://hireamir.com/blog/software-anti-patterns-magic-numbers